### DISCUSSION TITLE
Allow `hide` property to be set dynamically

### DISCUSSION THREAD

**Discussion by yannci** [elchininet/custom-sidebar#265] [Ideas]
First of all, thank you very much custom-sidebar and the recent improvements. Definitely a live safer when it comes to a multitude of dashboards!

The title might already give an idea of what I am after but to boil it down: I'd like to hide or unhide items in the sidebar based on entity states or best case scenario a template. Let's say we have an input_boolean `development mode` with which I'd like to drive the visibility of developer tools, automations ... you name it. Or show a certain dashboard only at a certain time.

I thought about various ways how to achieve this, exceptions, a partial or setting a style property but I don't think any of those would work if I am not mistaken?

Interested to hear what you think about it!

Cheers,
Yannic

**Comment by elchininet** [comment]
Hi @yannci,
Please, check [this comment](https://github.com/elchininet/custom-sidebar/discussions/163#discussioncomment-11120989). I'll let you know if at some point it is implemented, but your best option for now is to use [card-mod](https://github.com/thomasloven/lovelace-card-mod) for anything that the plugin doesn't offer natively.
Regards

**Reply by yannci** [reply]
Hey @elchininet,

Thanks for the quick response. I went through previous discussion but missed this. The explanation makes sense and I will see how far I get with card-mod or hack around it for the time being since I don't need the keyboard events. Good insight though and I appreciate your work here.

Cheers!

**Reply by elchininet** [reply]
Hi @yannci,
I have started that work multiple times but didn't have enough time to solve it. I have a branch with some work done, it is working so far but the keyboard navigation is broken. I need to check if I can solve the keyboard navigation if not I can compile that version without the keyboard navigation and submit it in a gist so you can download it and install manually.
I'll let you know.

**Reply by elchininet** [reply]
Hi @yannci,
Please, try [the last release](https://github.com/elchininet/custom-sidebar/releases/tag/v8.2.0) and report back if it works OK for you or if it creates any issues.
Regards

**Reply by yannci** [reply]
Hey,

I installed the new version yesterday and so far it works like a charm! I will report back if there will be any issues coming up.

Thanks for your work and the quick implementation. This really is a game changer for me!

Cheers,
Yannic

**Reply by elchininet** [reply]
@yannci,
Glad to hear that 👍
Cheers

**Comment by SEETechWolf** [comment]
Thanks for this feature it is great!
